contentcuration/contentcuration/tests/test_rest_framework.py

```diff
@@ -36,3 +42,84 @@ def test_readonly_fields(self):
         self.channel.refresh_from_db()
         self.assertEqual(original_version, self.channel.version)
+
+
+class AssessmentItemTestCase(BaseAPITestCase):
+
+    def test_bulk_update(self):
+        exercise = ContentNode.objects.filter(kind=content_kinds.EXERCISE).first()
+        item1 = AssessmentItem.objects.create(contentnode=exercise)
+        item2 = AssessmentItem.objects.create(contentnode=exercise)
+        item3 = AssessmentItem.objects.create(contentnode=exercise)
+        item1dict = {}
+        item2dict = {}
+        item3dict = {}
+        for field in AssessmentItem._meta.fields:
+            attname = field.attname
+            set_attname = attname
+            if attname == 'contentnode_id':
+                set_attname = 'contentnode'
+            item1dict[set_attname] = getattr(item1, attname)
+            item2dict[set_attname] = getattr(item2, attname)
+            item3dict[set_attname] = getattr(item3, attname)
+        item1dict["question"] = "test"
+        item2dict["type"] = "test"
+        self.client.put(reverse_lazy('assessmentitem-list'), json.dumps([item1dict, item2dict, item3dict]), content_type="application/json")
+        item1.refresh_from_db()
+        self.assertEqual(item1.question, 'test')
+        item2.refresh_from_db()
+        self.assertEqual(item2.type, 'test')
+        item3.refresh_from_db()
+        self.assertEqual(item3.question, item3dict['question'])
+
+    def test_bulk_update_non_existent_item(self):
+        exercise = ContentNode.objects.filter(kind=content_kinds.EXERCISE).first()
+        item1 = AssessmentItem.objects.create(contentnode=exercise)
+        item1dict = {}
+        item2dict = {}
+        item3dict = {}
+        for field in AssessmentItem._meta.fields:
+            attname = field.attname
+            set_attname = attname
+            if attname == 'contentnode_id':
+                set_attname = 'contentnode'
+            item1dict[set_attname] = getattr(item1, attname)
+            item2dict[set_attname] = getattr(item1, attname)
+            item3dict[set_attname] = getattr(item1, attname)
+        item2dict['id'] = 10000
+        item3dict['id'] = 10001
+        item1dict["question"] = "test"
+        response = self.client.put(reverse_lazy('assessmentitem-list'), json.dumps([item1dict, item2dict, item3dict]), content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_bulk_update_checksum_file_not_associated_create_new_file_object(self):
+        exercise = ContentNode.objects.filter(kind=content_kinds.EXERCISE).first()
+        item1 = AssessmentItem.objects.create(contentnode=exercise)
+        item1dict = {}
+        for field in AssessmentItem._meta.fields:
+            attname = field.attname
+            set_attname = attname
+            if attname == 'contentnode_id':
+                set_attname = 'contentnode'
```

Comment:
I think this change looks good to go, but seeing this block in each test is setting off my DRY-dar. ;-)

Reply:
Heh, I tend towards WETness in tests, unless you start repeating the same code many many times.